packages/amplify-category-function/src/provider-utils/awscloudformation/utils/permissionMapUtils.ts

```diff
@@ -8,6 +11,36 @@ export const fetchPermissionResourcesForCategory = (permissionMap, category: str
   return _.keys(_.get(permissionMap, [category]));
 };
 
-export const fetchPermissionsForResourceInCategory = (permissionMap, category: string, resourceName: string) => {
-  return _.get(permissionMap, [category, resourceName]);
+export const fetchPermissionsForResourceInCategory = (
+  permissionMap: object = {},
+  category: string,
+  resourceName: string,
+  serviceType: string,
+) => {
+  const permissions = _.get(permissionMap, [category, resourceName], []);
+  // If AppSync and legacy CRUD permissions BUT they opted into the feature flag, update policy.
+  if (serviceType == 'AppSync' && hasCRUDPermissions(permissions) && featureFlagPR5342()) {
+    return updatePermissionsForBackwardsCompatibility(permissionMap, category, resourceName);
```

Comment:
Why is the FF branch of the if doing anything that is backward compatible? It should not, right?

Reply:
I wrote it so that enabling a feature flag with incorrect permissions would autocorrect their policies. I now see that the way feature flags are being used in amplify is that by turning them on early, the user may deals with the consequences. I see this results in code with less 'artifacts' so I'll update to fit this design.

Reply:
I moved the FF check to line 102 of `execPermissionsWalkthrough.ts`. Still exists as an artifact in the code, but there is no more autocorrecting of permissions. Is this the intention?